### PR TITLE
[Fix] Bronjahm event & achievement Soul Power

### PR DESCRIPTION
**Pull request by ghost**
Based on PR #13606 without SQL code

[Initialize all fields in the constructor] Requested by @jackpoz in PR #13606
Solution delivered by @NorthStrider in PR #14036

``` c++
{
    DoCast(me, SPELL_SOULSTORM_CHANNEL, true);
    _soulFragmentsSpawned = 0;
}
```

Added indentations suggested by @Nyeriah to keep codestyle intact.
Added the following code suggested by NorthStrider in PR #14036 :
`if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, NonTankTargetSelector(me)))`

### CODE REVIEW

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 1]
```diff
@@ -21,13 +21,15 @@
...
-    SAY_AGGRO           = 0,
-    SAY_SLAY            = 1,
-    SAY_DEATH           = 2,
-    SAY_SOUL_STORM      = 3,
-    SAY_CORRUPT_SOUL    = 4,
+    EVENT_MAGIC_BANE      = 1,
+    EVENT_SHADOW_BOLT     = 2,
+    EVENT_CORRUPT_SOUL    = 3,
+    EVENT_SOULSTORM       = 4,
+    EVENT_FEAR            = 5
```
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: [**Nyeriah**](https://github.com/Nyeriah) added a note [19 hours ago](https://github.com/TrinityCore/TrinityCore/pull/14036#discussion_r24174111 "Feb. 5. 2015, 17:14 CET")
Did you check it in the database to see if these values match the data expected to be used? In the last author's PR it was wrong

Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: [**MitchesD**](https://github.com/MitchesD) added a note [17 hours ago](https://github.com/TrinityCore/TrinityCore/pull/14036#discussion_r24184413 "Feb. 5. 2015, 19:12 CET")
yeah it seems to be wrong too

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 2]
```diff
@@ -69,43 +70,43 @@ class boss_bronjahm : public CreatureScript
...
-                instance->SetBossState(DATA_BRONJAHM, IN_PROGRESS);
+                Talk(SAY_AGGRO);
             }
 
             void JustDied(Unit* /*killer*/) override
             {
-                Talk(SAY_DEATH);
+                _JustDied();
+                sCreatureTextMgr->SendSound(me, SOUND_DEATH, CHAT_MSG_MONSTER_YELL, 0, TEXT_RANGE_NORMAL, TEAM_OTHER, false);
+            }
```
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: [**Nyeriah**](https://github.com/Nyeriah) added a note [19 hours ago](https://github.com/TrinityCore/TrinityCore/pull/14036#discussion_r24174111 "Feb. 5. 2015, 17:14 CET")
Use creature_text here (the entry is already in the database)

**Comment by tkrokli** [comment]
The Travis CI build failed - I guess I need to test the code I was told is wrong...

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp [hunk 3]
```diff
@@ -114,7 +115,7 @@ class boss_bronjahm : public CreatureScript
                 {
                     events.SetPhase(PHASE_2);
                     DoCast(me, SPELL_TELEPORT);
-                    events.ScheduleEvent(EVENT_FEAR, urand(12000, 16000), 0, PHASE_2);
+                    events.ScheduleEvent(EVENT_FEAR, 6000, 0, PHASE_2);
```
Comment (MitchesD) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: do you have sniff confirmation for this?

**Comment by tkrokli** [comment]
Sorry, Git keeps deleting my current branch when I Push again to my repo.

**Comment by MitchesD** [comment]
Hey, stop doing stupid things please. Use some git extension. For example TortoiseGit or GitExtensions. Your code has many bad things and also I think it's not tested ingame. So, at first learn how to work with git, at least some basic things like pull, push, rebase, ..then create a TESTED Pull request with all those all changes we told to you.

File: src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp (hunk 3 above)
Comment (tkrokli) [inline src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp]: No. I don't have anything else than the old abandoned PR.

**Pull request closed without merging**